src/derivative_operators/concretization.jl

```diff
@@ -1,123 +1,60 @@
-function LinearAlgebra.Array(A::DerivativeOperator{T}, N::Int=A.len) where T
-    L = zeros(T, N, N+2)
+function Base.copyto!(L::AbstractMatrix{T}, A::DerivativeOperator{T}, N::Int) where T
     bl = A.boundary_point_count
     stencil_length = A.stencil_length
+    stencil_pivot = use_winding(A) ? (1 + stencil_length%2) : div(stencil_length,2)    
     bstl = A.boundary_stencil_length
+    
     coeff   = A.coefficients
-
-    if use_winding(A)
-        stencil_pivot = 1 + A.stencil_length%2
+    get_coeff = if coeff isa AbstractVector
+        i -> coeff[i]
+    elseif coeff isa Number
+        i -> coeff
     else
-        stencil_pivot = div(stencil_length,2)
+        i -> true
     end
 
     for i in 1:bl
-        cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
+        cur_coeff   = get_coeff(i)
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(A.low_boundary_coefs[i]) : A.low_boundary_coefs[i]
         L[i,1:bstl] = cur_coeff * cur_stencil
     end
 
     for i in bl+1:N-bl
-        cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
+        cur_coeff   = get_coeff(i)
         stencil     = eltype(A.stencil_coefs) <: AbstractVector ? A.stencil_coefs[i] : A.stencil_coefs
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(stencil) : stencil
         L[i,i+1-stencil_pivot:i-stencil_pivot+stencil_length] = cur_coeff * cur_stencil
     end
 
     for i in N-bl+1:N
-        cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
+        cur_coeff   = get_coeff(i)
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(A.high_boundary_coefs[i-N+bl]) : A.high_boundary_coefs[i-N+bl]
         L[i,N-bstl+3:N+2] = cur_coeff * cur_stencil
     end
-    return L
+    
+    L
 end
 
-function SparseArrays.SparseMatrixCSC(A::DerivativeOperator{T}, N::Int=A.len) where T
-    L = spzeros(T, N, N+2)
-    bl = A.boundary_point_count
-    stencil_length = A.stencil_length
-    stencil_pivot = div(stencil_length,2)
-    bstl = A.boundary_stencil_length
-    coeff   = A.coefficients
+LinearAlgebra.Array(A::DerivativeOperator{T}, N::Int=A.len) where T =
+    copyto!(zeros(T, N, N+2), A, N)
 
-    if use_winding(A)
-        stencil_pivot = 1 + A.stencil_length%2
-    else
-        stencil_pivot = div(stencil_length,2)
-    end
+SparseArrays.SparseMatrixCSC(A::DerivativeOperator{T}, N::Int=A.len) where T =
+    copyto!(spzeros(T, N, N+2), A, N)

```

Comment:
This would be really slow?

Reply:
Why would it? I overloaded `Base.copyto!` at the top of the file, the code of which is exactly the same as was in all three versions for different target matrices.

Reply:
Adding to a sparse matrix using indexing is really slow: the indices should all be declared up front.

Reply:
oh I see, ehh, it's not any worse than before, but it's still bad. If no one wants to actually do the `(I,J,V)`, then it's best to probably just build the banded matrix and then convert.

Reply:
Sure :) I was anyway interested in the banded versions, but could not keep my hands off refactoring this.

Reply:
Yeah, 99% of the time people should just call `sparse` and get the banded version.